### PR TITLE
Expose `User.reload` as beta API

### PR DESCRIPTION
**Pull request by jglick**
CloudBees CI in HA mode sometimes needs to reload the list of `User`s in response to changes on disk. #8544 suffices when there is merely a configuration change to a loaded `User` object (e.g., saving email address) but not in all cases, such as a new `User` being defined. Calling `Jenkins.reload` is much too heavyweight.

### Testing done

Covered by an automated test in CloudBees CI.

### Proposed changelog entries

- N/A

### Proposed upgrade guidelines

N/A

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by Vlatombe — APPROVED** [review]

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
